src/views/PrimaryTypeNode.ts

```diff
@@ -121,6 +121,11 @@ export class PrimaryTypeNode extends DataNode {
             contextValue += "+test";
         }
 
+        if (this._rootNode?.getParent() instanceof ProjectNode
+            && (this._rootNode.getParent() as ProjectNode).nodeData?.metaData?.MaxSourceVersion >= 14) {
```

Comment:
Given that record is a preview feature in Java 14 and become a standard feature starting from Java 16. I think we should compare with `16` here.

Reply:
yes, makes sense. The latest Java extension doesn't support preview for the old versions like 14 and 15, it makes sense to enable it only on Java 16 and higher.

Reply:
yes, makes sense